### PR TITLE
Add Storybook for displaying demos and UI-kit

### PR DESCRIPTION
**Pull request by bryceosterhaus**
Hey all, thought I would send a POC for this. The idea is to use storybook for demos and it would actually allow us to build a static site for documentation.

Here is a demo of both badge and button on storybook. Checkout the ClayButton default story and play around with the "knobs" tab. Thought it was pretty cool that it'd allow us to interactively play with our components. https://storybook-static-apenpbqdiu.now.sh/?path=/story/claybadge--default

If we do decide this is the route we want to go, I will update the PR to add stories for all of our existing components and I will remove our `demo` directories.

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by matuzalemsteles** [comment]
hey @bryceosterhaus the experience is very good for development. By me we can remove the demos and just keep the Storybook. This will add much more to our documentation.

It's a perfect replacement for our current demos, since we have the a11y addon and also the demos get much more organized and easy to navigate. Now would it be possible to run a11y tests on CI?

**Comment by bryceosterhaus** [comment]
I also investigated using netlify to auto deploy storybook builds. Seems pretty cool. We would be able to automatically have static sites generated for every PR, which would allow quick user testing. We can also automate it so that each branch or tag could have a build and then set a custom domain.

**Comment by matuzalemsteles** [comment]
That sounds great, it's something we want to do with clayui.com (#1170).

**Comment by bryceosterhaus** [comment]
If no one has any issues with using storybook, I'm gonna go ahead and replace all of our existing demos with this. We can decide on the auto-deployment portion later.

**Comment by matuzalemsteles** [comment]
hey @bryceosterhaus, I think you can go with it, I see no reason why we will not add the Storybook.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated to add all demos. I also had to modify some file structure and config so that the modules properly built. See https://github.com/liferay/clay/issues/1744 for reference

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-sticker/package.json [hunk 1]
```diff
@@ -8,27 +8,34 @@
...
+	"keywords": [
+		"clay",
+		"react"
+	],
 	"dependencies": {
 		"classnames": "^2.2.6"
 	},
 	"devDependencies": {
-		"@clayui/icon": "^3.0.0"
+		"@clayui/icon": "v3.0.0"
```
Comment (matuzalemsteles) [inline packages/clay-sticker/package.json]: Should this be `v3.0.0` anyway?

Comment (bryceosterhaus) [reply packages/clay-sticker/package.json]: good catch, just updated.

**Pull request merged**